iceoryx_dds_gateway/CMakeLists.txt

```diff
@@ -0,0 +1,95 @@
+cmake_minimum_required(VERSION 3.7)
+
+set(iceoryx_dds_gateway_VERSION 0.1.0)
+project(iceoryx_dds_gateway VERSION ${iceoryx_dds_gateway_VERSION})
+
+option(BUILD_TESTS "build test binaries")
+
+find_package(iceoryx_posh REQUIRED)
+find_package(iceoryx_utils REQUIRED)
+
+include(IceoryxPackageHelper)
+
+#
+########## set variables for library export ##########
+#
+setup_package_name_and_create_files(
+    NAME ${PROJECT_NAME}
+    NAMESPACE iceoryx_dds_gateway
+    PROJECT_PREFIX ${PREFIX}
+)
+
+#
+########## feature flags ##########
+#
+option(USE_CYCLONE_DDS "Bind to CycloneDDS implementation" on)
+
+if(USE_CYCLONE_DDS)
+    message(INFO " Using CycloneDDS stack")
+    find_package(CycloneDDS CONFIG REQUIRED)
+    find_package(CycloneDDS_CXX_API CONFIG REQUIRED)
+    find_package(Idlpp-cxx CONFIG REQUIRED)
+endif()
+
+#
+########## build building-block library ##########
+#
+add_library(ioxdds
+    STATIC
+    source/ioxdds/log/logging.cpp
+)
+add_library(${PROJECT_NAMESPACE}::ioxdds ALIAS ioxdds)
```

Comment:
this is a little bit inconsistent with iceoryx_posh and iceoryx_utils, where the main library is called like the module
@mossmaurice what do you think?

Reply:
Yeah, I think `iceoryx_dds` would be nicer. Maybe something we can add to #109?

Reply:
Sounds good to me! This was kind of a placeholder name anyway. I will change it in the next PR (and add to the mentioned issue).